virttest/utils_libvirt/libvirt_vmxml.py

```diff
@@ -148,10 +155,13 @@ def modify_vm_device(vmxml, dev_type, dev_dict=None, index=0, virsh_instance=vir
 
         vmxml.devices = xml_devices
         vmxml.xmltreefile.write()
-        vmxml.sync(virsh_instance=virsh_instance)
+        if need_sync:
+            vmxml.sync(virsh_instance=virsh_instance)
     except IndexError:
         dev_obj = create_vm_device_by_type(dev_type, dev_dict)
-        libvirt.add_vm_device(vmxml, dev_obj, virsh_instance=virsh_instance)
+        vmxml = libvirt.add_vm_device(vmxml, dev_obj,
+                                      virsh_instance=virsh_instance,
+                                      need_sync=need_sync)
 
     LOG.debug(f"XML of {dev_type} device is:\n{dev_obj}")
-    return dev_obj
+    return dev_obj, vmxml
```

Comment:
Besides, I don't think we need to return `vmxml` in this situation since `vmxml` has been updated after `vmxml.xmltreefile.write()`.  
So we could get what we want simply by:
```
vmxml = .....
libvirt_vmxml.modify_vm_device(vmxml, .....
```
Then vmxml should be updated

Reply:
+1 @chloerh

Reply:
Updated, thanks